addons/remember-collapsed-categories/userscript.js

```diff
@@ -0,0 +1,56 @@
+export default async function ({ console, addon }) {
+  while (true) {
+    const categoryBody = await addon.tab.waitForElement('[id^="category_body_"]', {
+      markAsSeen: true,
+    });
+    console.log(`Found category ${categoryBody.id}`);
+
+    const categoryCollapse = categoryBody.querySelector(".toggle");
+    categoryCollapse.addEventListener("click", (event) => {
+      event.preventDefault();
+      const categoryHead = event.target.parentElement;
+      categoryHead.classList.toggle("sa-collapsed");
+      if (categoryHead.classList.contains("sa-collapsed")) {
+        console.log(`${categoryHead.id.split("_")[2]} is being opened`);
+        return;
+      } else {
+        extendCategoryCookieExistenceLength(categoryNumber);
+      }
+    });
+
+    const categoryNumber = categoryBody.id.split("_").pop();
+    if (!shouldCollapseCategory(categoryNumber)) {
+      removeCategoryCookie(categoryNumber);
+      continue;
+    }
+    extendCategoryCookieExistenceLength(categoryNumber);
+
+    const categoryContent = categoryBody.querySelector(".box-content");
+    categoryContent.style.display = "none";
+
+    const categoryHead = categoryBody.querySelector(`.box-head h4`);
+    categoryHead.classList.add("collapsed");
+    categoryHead.classList.add("sa-collapsed");
```

Comment:
Does Scratch style category heads that have the "collapsed" class any different?

Does the "sa-collapsed" class have any specific purpose? I was expecting a userstyle targeting this class name, but there is none,

Reply:
I think there was a reliability issue with `collapsed`, if I'm remember correctly from initially doing that.